src/pages/SearchPage.js

```diff
@@ -45,69 +45,68 @@ const defaultProps = {
     reports: {},
 };
 
-class SearchPage extends Component {
-    constructor(props) {
-        super(props);
+function SearchPage(props) {
+    const {recentReports: initialRecentReports, personalDetails: initialPersonalDetails, userToInvite: initialUserToInvite} = OptionsListUtils.getSearchOptions(props.reports, props.personalDetails, '', props.betas);
 
-        Timing.start(CONST.TIMING.SEARCH_RENDER);
-        Performance.markStart(CONST.TIMING.SEARCH_RENDER);
+    const [searchValue, setSearchValue] = useState('')
+    const [recentReports, setRecentReports] = useState(initialRecentReports)
+    const [personalDetails, setPersonalDetails] = useState(initialPersonalDetails)
+    const [userToInvite, setUserToInvite] = useState(initialUserToInvite)
 
-        this.searchRendered = this.searchRendered.bind(this);
-        this.selectReport = this.selectReport.bind(this);
-        this.onChangeText = this.onChangeText.bind(this);
-        this.debouncedUpdateOptions = _.debounce(this.updateOptions.bind(this), 75);
+    const updateOptions = useCallback(() => {
+        const {recentReports: localRecentReports, personalDetails: localPersonalDetails, userToInvite: localUserToInvite} = OptionsListUtils.getSearchOptions(
+            props.reports,
+            props.personalDetails,
+            searchValue.trim(),
+            props.betas,
+        );
 
-        const {recentReports, personalDetails, userToInvite} = OptionsListUtils.getSearchOptions(props.reports, props.personalDetails, '', props.betas);
+        setUserToInvite(localUserToInvite)
+        setRecentReports(localRecentReports)
+        setPersonalDetails(localPersonalDetails)
+    }, [props.reports, props.personalDetails, searchValue, props.betas])
 
-        this.state = {
-            searchValue: '',
-            recentReports,
-            personalDetails,
-            userToInvite,
-        };
-    }
+    const debouncedUpdateOptions =_.debounce(updateOptions, 75);
```

Comment:
why this? if the `updateOptions` is getting call very frequently, it is something wrong with the logic, we shouldn't debounce

Reply:
Yes, the app should have the same reference across the re-renders. useCallback is necessary here.